# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/base/AbstractRange.cpp
# Commit: fb611166c7ab
# Full Hash: fb611166c7ab7d3c67f2361ec8c45e724d2e6dcc
# Author: Sean Feng <sefeng@mozilla.com>
# Date: 2025-05-09 04:15:00
# Regressor Bug: 1932150
# File Overlap Count: 2
# Description:
#   Bug 1932150 - Fix some inconsistency about indicating whether the selection is allowed to cross the shadow boundary r=jjaschke,dom-core
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D246949
# ==============================================================================

diff -r 53dc4e16ebe8 -r fb611166c7ab dom/base/AbstractRange.cpp
--- a/dom/base/AbstractRange.cpp	Thu May 08 12:56:29 2025 +0000
+++ b/dom/base/AbstractRange.cpp	Thu May 08 13:37:20 2025 +0000
@@ -249,14 +249,10 @@
   if (!mIsPositioned) {
     return nullptr;
   }
-  nsINode* startContainer =
-      aAllowCrossShadowBoundary == AllowRangeCrossShadowBoundary::Yes
-          ? GetMayCrossShadowBoundaryStartContainer()
-          : GetStartContainer();
-  nsINode* endContainer =
-      aAllowCrossShadowBoundary == AllowRangeCrossShadowBoundary::Yes
-          ? GetMayCrossShadowBoundaryEndContainer()
-          : GetEndContainer();
+  nsINode* startContainer = ShadowDOMSelectionHelpers::GetStartContainer(
+      this, aAllowCrossShadowBoundary);
+  nsINode* endContainer = ShadowDOMSelectionHelpers::GetEndContainer(
+      this, aAllowCrossShadowBoundary);
 
   if (MayCrossShadowBoundary() &&
       aAllowCrossShadowBoundary == AllowRangeCrossShadowBoundary::Yes) {